test/LogTest.php

```diff
@@ -0,0 +1,222 @@
+ * Foundation; either version 2 of the Licence, or any later version.
+ * OPUS is distributed in the hope that it will be useful, but WITHOUT ANY
+ * WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU General Public License for more
+ * details. You should have received a copy of the GNU General Public License
+ * along with OPUS; if not, write to the Free Software Foundation, Inc., 51
+ * Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+ *
+ * @category    Test
+ * @package     OpusTest
+ * @author      Kaustabh Barman <barman@zib.de>
+ * @copyright   Copyright (c) 2020, OPUS 4 development team
+ * @license     http://www.gnu.org/licenses/gpl.html General Public License
+ */
+
+namespace OpusTest;
+
+use Opus\Log;
+
+class LogTest extends \PHPUnit_Framework_TestCase
+{
+
+    private $logFile;
+
+    public function tearDown()
+    {
+        if (is_resource($this->logFile)) {
+            fclose($this->logFile);
+        }
+
+        parent::tearDown();
+    }
+
+    public function testSetLevel()
+    {
+        $opusLog = $this->getOpusLog();
+        $opusLog->setLevel(\Zend_Log::DEBUG);
+        $debugMessage = 'Debug level message from testSetLevel';
+        $opusLog->debug($debugMessage);
+        $content = $this->readLog();
+
+        $level = $opusLog->getLevel();
+
+        $this->assertContains($debugMessage, $content);
+        $this->assertEquals(\Zend_Log::DEBUG, $level);
+    }
+
+    public function testSetLevelForFilterDisabling()
+    {
+        $opusLog = $this->getOpusLog();
+        $opusLog->setLevel(null);
+
+        $debugMessage = 'Debug level message from testSetLevelWithNullLevel';
+        $opusLog->debug($debugMessage);
+
+        $content = $this->readLog();
+
+        $this->assertContains($debugMessage, $content);
+    }
+
+    public function testSetLevelToNullHavingCustomLevels()
+    {
+        $opusLog = $this->getOpusLog();
+        $opusLog->addPriority('TEST', 8);
+        $opusLog->setLevel(null);
+        $opusLog->addPriority('TLEVEL', 9);
+
+        $testMessage = 'Test level message';
+        $opusLog->test($testMessage);
+
+        $tlevelMessage = 'Tlevel message';
+        $opusLog->tlevel($tlevelMessage);
+
+        $content = $this->readLog();
+
+        $this->assertContains($testMessage, $content);
+        $this->assertContains($tlevelMessage, $content);
+    }
+
+    public function testSetLevelNotEffectingOtherFilters()
+    {
+        $opusLog = $this->getOpusLog();
+
+        $filter = new \Zend_Log_Filter_Priority(\Zend_Log::WARN);
+        $opusLog->addFilter($filter);
+
+        $infoMessage = 'Info level Message';
+        $opusLog->info($infoMessage);
+
+        $opusLog->setLevel(\Zend_Log::NOTICE);
+        $noticeMessage = 'Notice Level Message';
+        $opusLog->notice($noticeMessage);
+
+        $opusLog->setLevel(\Zend_Log::ERR);
```

Comment:
I think this test is really hard to read and an additional level change here does add more information. Please try to make this easier to understand.

Reply:
Maybe even add some brief comments (`//`).